### PR TITLE
mark image input source as experimental

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton @csrwng ptal

File: pkg/build/api/types.go [hunk 1]
```diff
@@ -189,6 +189,8 @@ type BuildSource struct {
 	Git *GitBuildSource
 
 	// Image describes an image to be used to provide source for the build
+	// EXPERIMENTAL.  This will be changing to an array of images in the near future
+	// and no migration/compatibility will be provided.  Use as your own risk.
```
Comment (csrwng) [inline pkg/build/api/types.go]: s/as/at/

Comment (csrwng) [inline pkg/build/api/types.go]: same below

File: pkg/build/api/v1/types.go [hunk 2]
```diff
@@ -161,6 +161,8 @@ type BuildSource struct {
 	Git *GitBuildSource `json:"git,omitempty" description:"optional information about git build source"`
 
 	// Image describes an image to be used to provide source for the build
+	// EXPERIMENTAL.  This will be changing to an array of images in the near future
+	// and no migration/compatibility will be provided.  Use as your own risk.
 	Image *ImageSource `json:"image,omitempty" description:"optional image build source"`
```
Comment (deads2k) [inline pkg/build/api/v1/types.go]: Please include the warning in the description, so it ends up in swagger.

Comment (csrwng) [inline pkg/build/api/v1/types.go]: https://github.com/openshift/origin/blob/master/api/definitions/v1.imagesource/description.adoc

**Comment by bparees** [comment]
@csrwng @deads2k fixed.

**Comment by csrwng** [comment]
lgtm

File: pkg/build/api/v1/types.go [hunk 3]
```diff
@@ -161,6 +161,8 @@ type BuildSource struct {
 	Git *GitBuildSource `json:"git,omitempty" description:"optional information about git build source"`
 
 	// Image describes an image to be used to provide source for the build
+	// EXPERIMENTAL.  This will be changing to an array of images in the near future
+	// and no migration/compatibility will be provided.  Use at your own risk.
 	Image *ImageSource `json:"image,omitempty" description:"optional image build source"`
```
Comment (deads2k) [inline pkg/build/api/v1/types.go]: Do you mind adding it to the description tag on this line so that `api/swagger-spec/oapi-v1.json` is updated.

Comment (bparees) [inline pkg/build/api/v1/types.go]: nope, good call.

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
@deads2k woot!

**Pull request merged**